source/includes/_node_rpc_mining.md

```diff
@@ -0,0 +1,618 @@
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  // Block data is old, so it should return error
+  const res = await rpc.execute('verifyblock', [ blockdata ]);
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+Verifies the block data.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+1  | blockdata | Required. | Mined block data (hex)
+
+
+
+
+
+## setgenerate
+
+```javascript
+let mining, proclimit;
+```
+
+```shell--vars
+mining=1;
+proclimit=1;
+```
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "setgenerate",
+    "params": [ "'$mining'", "'$proclimit'" ]
+  }'
+```
+
+```shell--cli
+bcoin cli rpc setgenerate $mining $proclimit
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  const res = await rpc.execute('setgenerate', [ mining, proclimit ]);
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+true
+```
+
+Will start the mining on CPU.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+1  | mining | 0 | `true` will start mining, `false` will stop.
+2 | proclimit | 0 | 
+
+
+
+## getgenerate
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "getgenerate",
+    "params": []
+  }'
+```
+
+```shell--cli
+bcoin cli rpc getgenerate
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  const res = await rpc.execute('getgenerate');
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+true
+```
+
+Returns status of mining on Node.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+None.
+
+
+
+## generate
+
+```javascript
+let numblocks;
+```
+
+```shell--vars
+numblocks=2;
+```
+
+```shell--curl
+# Will return once all blocks are mined.
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "generate",
+    "params": [ "'$numblocks'" ]
+  }'
+```
+
+```shell--cli
+# Timeout error
+bcoin cli rpc generate $numblocks
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  // Timeout error
+  const res = await rpc.execute('generate', [ numblocks ]);
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+[
+  "0aac2269ce67d5bda3743b1a179a7824f17b3b6df011befd7bd224cb958bd4ec",
+  "0871f1a42279397b7508c75e142b903a945a19af08c6de6970e266255d10f08f"
+]
+```
+Mines `blocknumber` blocks.
```

Comment:
should this be "Mines `numblocks` number of blocks"? It looks like `blocknumber` is supposed to be the name of the param.